compiler/modules/nback/scaffold.bal

```diff
@@ -471,8 +471,10 @@ function semTypeRepr(t:SemType ty) returns Repr {
         panic err:impossible("allocate register with never type");
     }
     int supported = t:NIL|t:BOOLEAN|t:INT|t:FLOAT|t:STRING|t:LIST|t:MAPPING|t:ERROR;
-    // DECIMAL is here for ConvertToInt or ConvertToFloat 
-    if (w | supported | t:DECIMAL) == t:TOP || (w & supported) == w {
+    // DECIMAL is here for ConvertToInt or ConvertToFloat
+    // TABLE and XML are here for anydata
+    int maximized = w | supported | t:DECIMAL | t:TABLE | t:XML;
+    if maximized == t:TOP || maximized == (t:NON_BEHAVIORAL|t:ERROR) {
```

Comment:
My suggestion was:

```
int maximized = w | supported | t:DECIMAL;
if maximized == t:TOP || maximized == t:NON_BEHAVIOURAL|t:ERROR ...
```

which is not the same.  You need to call out explicitly if something I suggest doesn't work.

The `...` meant this:

```
int maximized = w | supported | t:DECIMAL;
if maximized == t:TOP || maximized == t:NON_BEHAVIOURAL|t:ERROR || (w & supported) == w { ... }
```

Reply:
fixed